e2e/testcases/basic_test.go

```diff
@@ -0,0 +1,365 @@
+	nt.RootRepos[configsync.RootSyncName].Copy("../../examples/acme/namespaces/eng/backend/bob-rolebinding.yaml", "acme/namespaces/eng/backend/br.yaml")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Add bob rolebinding")
+	nt.WaitForRepoSyncs()
+	if err := nt.Validate("bob-rolebinding", "backend", &rbacv1.RoleBinding{}, RoleBindingsHasName("acme-admin")); err != nil {
+		nt.T.Fatal("bob-rolebinding not found")
+	}
+
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/robert-rolebinding.yaml", yamlDir), "acme/namespaces/eng/backend/br.yaml")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Replace bob with robert rolebinding")
+	nt.WaitForRepoSyncs()
+
+	if err := nt.ValidateNotFound("bob-rolebinding", "backend", &rbacv1.RoleBinding{}); err != nil {
+		nt.T.Fatal("bob-rolebinding is not deleted")
+	}
+
+	if err := nt.Validate("robert-rolebinding", "backend", &rbacv1.RoleBinding{}); err != nil {
+		nt.T.Fatal("robert-rolebinding not found")
+	}
+}
+
+func manageNamespace(nt *nomostest.NT, namespace string) {
+	nt.T.Log("Add an unmanaged resource into the namespace as a control")
+	nt.T.Log("We should never modify this resource")
+	_, err := nt.Kubectl("apply", "-f", fmt.Sprintf("%s/reserved_namespaces/unmanaged-service.%s.yaml", yamlDir, namespace))
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Add resource to manage")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/reserved_namespaces/service.yaml", yamlDir), fmt.Sprintf("acme/namespaces/%s/service.yaml", namespace))
+	nt.T.Log("Start managing the namespace")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/reserved_namespaces/namespace.%s.yaml", yamlDir, namespace), fmt.Sprintf("acme/namespaces/%s/namespace.yaml", namespace))
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Start managing the namespace")
+	nt.WaitForRepoSyncs()
+
+	nt.T.Log("Validate managed service appears on the cluster")
+	if err := nt.Validate("some-service", namespace, &corev1.Service{}); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Remove the namespace directory from the repo")
+	nt.RootRepos[configsync.RootSyncName].Remove(fmt.Sprintf("acme/namespaces/%s", namespace))
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Remove the namespace from the managed set of namespaces")
+	nt.WaitForRepoSyncs()
+
+	nt.T.Log("Validate managed service disappears from the cluster")
+	if err := nt.ValidateNotFound("some-service", namespace, &corev1.Service{}); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Ensure that the unmanaged service remained")
+	if err := nt.Validate("some-other-service", namespace, &corev1.Service{}); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	someOtherService := &corev1.Service{}
+	if err := nt.Get("some-other-service", namespace, someOtherService); err != nil {
+		nt.T.Fatal(err)
+	}
+	if err := nt.Delete(someOtherService); err != nil {
```

Comment:
I suggest moving this up to a `Cleanup` block after this is created on line `286`

Reply:
moved inside a cleanup block